packages/testcontainers/src/container-runtime/clients/container/docker-container-client.ts

```diff
@@ -173,21 +174,31 @@ export class DockerContainerClient implements ContainerClient {
     }
   }
 
-  async exec(container: Container, command: string[], opts?: { log: boolean }): Promise<ExecResult> {
-    const chunks: string[] = [];
+  async exec(container: Container, command: string[], opts?: Partial<ExecOptions>): Promise<ExecResult> {
+    const execOptions: ExecCreateOptions = {
+      Cmd: command,
+      AttachStdout: true,
+      AttachStderr: true,
+    };
+
+    if (opts?.env !== undefined) {
+      execOptions.Env = Object.entries(opts.env).map(([key, value]) => `${key}=${value}`);
+    }
+    if (opts?.workingDir !== undefined) {
+      execOptions.WorkingDir = opts.workingDir;
+    }
+    if (opts?.user !== undefined) {
+      execOptions.User = opts.user;
+    }
```

Comment:
Not a blocker: have you checked how this null/undefined checks are done elsewhere in the library?

Just to confirm the evaluations are done consistently in the project.

Other than that. The ifs are fine for me.

Reply:
Sorry, I am not that familiar with the codebase style. 

But, for some context, the issue here boils down to what Javascript (and Typescript) understand as a "falsy" value (see [this](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)).